checkov/common/parallelizer/parallel_runner.py

```diff
@@ -19,46 +20,88 @@
 
 class ParallelRunner:
     def __init__(
-        self, workers_number: int | None = None, parallelization_type: ParallelizationType = ParallelizationType.FORK
+        self, workers_number: int | None = None
```

Comment:
The root issue is that SCA and SAST are not working well in Darwin via Fork mode (security issue of Mac).
There are multiple options to solve this:
1. **use Spawn** (default of Windows) - this is cross-platform solution but it erase the memory of each new process (not like fork the copy the parent process memory) - but to make it works need to go all over the parents memory and pass it to the child processes, it can take lots of time, I do see some attempt to support it:

```
            if parallel_runner.type == ParallelizationType.SPAWN:
                platform_integration_data = bc_integration.generate_instance_data()
```

2. **in case of macOS (Darwin) use threads** - this option is less isolated, and has less performance because of the GIL, but it will solve this issue.

3. **in case of macOS (Darwin) use threads only for SCA and SAST runners** - same as above, but the fork issue can happen in the future for other runners.

In my opinion it should be 2, to support all macOs cases, 
WDYT?